x/ecocredit/base/keeper/features/msg_add_allowed_bridge_chain.feature

```diff
@@ -0,0 +1,44 @@
+Feature: Msg/AddAllowedBridgeChain
+
+  A chain can be added to the list of allowed bridge chains:
+  - when the authority is the governance account
+  - when the chain name doesn't already exist in the state
+
+
+  Rule: The authority address is the governance account
+
+    Background:
+      Given the authority address "regen1nzh226hxrsvf4k69sa8v0nfuzx5vgwkczk8j68"
+
+    Scenario: the authority is the governance account
+      When alice attempts to add allowed bridge chain with properties
+      """
+      {
+        "authority": "regen1nzh226hxrsvf4k69sa8v0nfuzx5vgwkczk8j68",
+        "chain_name": "polygon"
+      }
+      """
+      Then expect no error
+
+    Scenario: the authority is not the governance account
+      When alice attempts to add allowed bridge chain with properties
+      """
+      {
+        "authority": "regen1fua8speyxgempgy06gpfs0p4z32zznkqakm57s",
+        "chain_name": "polygon"
+      }
+      """
+      Then expect the error contains "invalid authority"
+
+
+  Rule: The chain name must not already exist
+
+    Scenario: there is no chain name in state
+      When  alice attempts to add chain name "polygon"
+      Then expect no error
+      And expect chain name "POLYGON" to exist
```

Comment:
Any reason this is stored in all caps?

Reply:
the governance method itself enforces all caps. It could be all lower, just decided to go with caps. This way if someone made a proposal with "Polygon", "polygon", or "POLYGON", it would pass either way.

Reply:
related: https://github.com/cosmos/cosmos-sdk/issues/13129

Reply:
Hmmm.... I would prefer we sanitize and store all lowercase but maybe it's not a big deal. Might be confusing to reference as "polygon" and then "POLYGON" in the examples here.

Reply:
[refactor: use lowercase](https://github.com/regen-network/regen-ledger/pull/1462/commits/19d15cecb18e7a519ebc5a6a7fab66cdad646942)